### PR TITLE
feat: support indexing Tuple parameters, add tests

### PR DESCRIPTION
**Pull request by AayushSabharwal**
## Checklist

- [ ] Appropriate tests were added
- [ ] Any code changes were done in a way that does not break public API
- [ ] All documentation related to code changes were updated
- [ ] The new code follows the
  [contributor guidelines](https://github.com/SciML/.github/blob/master/CONTRIBUTING.md), in particular the [SciML Style Guide](https://github.com/SciML/SciMLStyle) and
  [COLPRAC](https://github.com/SciML/COLPRAC).
- [ ] Any new documentation only uses public API

### CODE REVIEW

**Comment by SebastianM-C** [comment]
When would one want to use `Tuple` indexing? Would that be used in conjunction with `MTKParameters`?

File: src/remake.jl [hunk 1]
```diff
@@ -19,14 +19,30 @@ function remake_buffer(sys, oldbuffer::AbstractArray, vals::Dict)
...
+        setu(sys, collect(keys(vals)))(newbuffer, elT.(values(vals)))
     else
         mutbuffer = remake_buffer(sys, collect(oldbuffer), vals)
         newbuffer = similar_type(oldbuffer, eltype(mutbuffer))(mutbuffer)
     end
     return newbuffer
 end
+
+mutable struct TupleRemakeWrapper
+    t::Tuple
```
Comment (ChrisRackauckas) [inline src/remake.jl:35]: This won't infer, Tuple isn't concrete

Comment (ChrisRackauckas) [inline src/remake.jl:35]:
```suggestion
mutable struct TupleRemakeWrapper{T <: Tuple}
    t::T
```

?

**Comment by ChrisRackauckas** [comment]
Let's follow up with the performance bit if it's needed.

File: src/remake.jl (hunk 1 above)
Comment (AayushSabharwal) [reply src/remake.jl:35]: The type of the tuple can change, which is why it's not fully typed

**Comment by AayushSabharwal** [comment]
This can't really be fully inferred, since
1. The dict in the last argument is not concrete
2. The type of the buffer can change

**Comment by ChrisRackauckas** [comment]
Gotcha

**Pull request merged**